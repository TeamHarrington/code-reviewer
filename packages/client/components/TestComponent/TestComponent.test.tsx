import { TestComponent } from '.'
import renderer from 'react-test-renderer'
import React from 'react'

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <TestComponent
      phrase="Hello World"
      href="https://google.com"
      onMouseEnter={() => undefined}
      onMouseLeave={() => undefined}
      _target="blank"></TestComponent>
  )
  let tree = component.toJSON()!
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onMouseEnter()
  // re-rendering
  tree = component.toJSON()!
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onMouseLeave()
  // re-rendering
  tree = component.toJSON()!
  expect(tree).toMatchSnapshot()
})
