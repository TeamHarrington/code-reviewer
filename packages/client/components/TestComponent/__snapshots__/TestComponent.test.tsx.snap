// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  _target="blank"
  className="TestComponent__StyledButton-sc-1bv19hd-0 leVzQg"
  href="https://google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Hello World
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  _target="blank"
  className="TestComponent__StyledButton-sc-1bv19hd-0 leVzQg"
  href="https://google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Hello World
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  _target="blank"
  className="TestComponent__StyledButton-sc-1bv19hd-0 leVzQg"
  href="https://google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Hello World
</a>
`;
